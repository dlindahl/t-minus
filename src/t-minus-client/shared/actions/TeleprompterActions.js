import TeleprompterActions from '../actionTypes/TeleprompterActionTypes'

export function setTeleprompterValue (value) {
  return {
    payload: { value },
    type: TeleprompterActions.TELEPROMPTER_VALUE_CHANGED
  }
}
