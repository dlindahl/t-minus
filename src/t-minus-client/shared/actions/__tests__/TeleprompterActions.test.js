/* eslint no-magic-numbers: off */
import * as teleprompter from '../TeleprompterActions'
import configureStore from 'redux-mock-store'

const mockStore = configureStore()

describe('TeleprompterActions', () => {
  describe('#setTeleprompterValue', () => {
    it('indicates that the teleprompter value has changed', () => {
      const store = mockStore()
      store.dispatch(teleprompter.setTeleprompterValue('VALUE'))
      const actions = store.getActions()
      expect(actions.length).toBe(1)
      expect(actions[0]).toMatchSnapshot()
    })
  })
})
