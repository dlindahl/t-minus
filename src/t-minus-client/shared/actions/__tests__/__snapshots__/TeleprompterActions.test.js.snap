exports[`TeleprompterActions #setTeleprompterValue indicates that the teleprompter value has changed 1`] = `
Object {
  "payload": Object {
    "value": "VALUE",
  },
  "type": "TELEPROMPTER_VALUE_CHANGED",
}
`;
